src/abstract.js

```diff
@@ -711,7 +711,7 @@ Sk.abstr.objectFormat = function (obj, format_spec) {
     // Find the (unbound!) __format__ method (a borrowed reference)
     meth = Sk.abstr.lookupSpecial(obj, Sk.builtin.str.$format);
     if (meth == null) {
-        throw new Sk.builtin.TypeError("Type " + Sk.abstr.typeName(obj) + " doesn't define __format__");
+        return Sk.misceval.callsimArray(Sk.builtin.object.prototype["__format__"], [obj, format_spec]);
```

Comment:
Why do we need this here?  Shouldn't lookupSpecial have found it?  This feels like a step backwards to have to look it up on the prototype after lookupSpecial...  I thought we had moved towards not doing this in new places?

Reply:
It's to handle old-style classes in Python 2, which do not inherit from `object`.

Reply:
Just to confirm, in Python 3, ```lookupSpecial``` will always succeed, right?

Reply:
Yep!